plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/flexible/FlexDeployCommandHandler.java

```diff
@@ -39,14 +46,22 @@ protected DeployPreferencesDialog newDeployPreferencesDialog(Shell shell, IProje
   }
 
   @Override
-  protected StagingDelegate getStagingDelegate(IProject project) {
+  protected StagingDelegate getStagingDelegate(IProject project) throws CoreException {
     String appYamlPath = new FlexDeployPreferences(project).getAppYamlPath();
     IFile appYaml = project.getFile(appYamlPath);
     if (!appYaml.exists()) {
```

Comment:
aha! This should not be a runtime exception since it indicates a problem outside the code itself. Separate bug but we should fix it.